docs/langref/relay/intro.rst

```diff
@@ -0,0 +1,205 @@
+==================
+Introduction
+==================
+
+Relay is differentiable programming language with support for
+closures, control-flow, and recursion. It has an advanced
+static type system specifically designed for programs written
+by machine  learning practioniers and researchers.
+
+Relay is intended to replace the computation graph based
+intermediate representations currently employed by deep
+learning frameworks and compilers. The deep learning community
+has organically evolved a representation that was useful
+for the form of computation originally desired, i.e
+a directed acyclic graph of primitive functions.
+
+This may be a useful way to describe early ML models, but is
+limited onceyou want to generalize to dynamic models.
+
+At the same time computation graphs have been over loaded
+purpose as both a compile-time and run-time data structure.
+The conflation of the description of a computation, the
+representation used for optimizing it, and the data structure
+used to execute it uncessarily hampers many of the goals
+of a framework.
+
+We believe having a high level, an expressive language designed
+for compiler optimizations is essential to the future of an
+end-to-end deep learning compiler stack.
+
+Relay's design was influenced by the authors previous experince
+building advanced optimizing compilers for high level languages,
+as well as experinces with the current TVM stack, and NNVM.
+
+Concretely NNVM has played the role of this high level IR.
+
+for NNVM which address design flaws of the computation graph IRs
+employed by common deep learning frameworks.
+We address a few important challenges. First we present a new IR with first
+class functions and closures, enabling the design of differentiable
+arbitrary control flow constructs. Second we contribute a type system for
+Relay which makes the important properties of computations involving tensors
+explicit.
```

Comment:
Miscellaneous small changes:
* "first-class" in this case should be hyphenated (compound adjective)
* There should be a comma in "the design of differentiable, arbitrary control flow constructs"
* There should be commas after "First" and "Second"
* It might be clearer in the last sentence if "makes the important properties of computations involving tensors
explicit" were "makes explicit the important properties of computations involving tensors"